File: 九、奇淫巧计类/7.实现CMD下载时的进度显示.go

package main

import (
	"fmt"
	"time"
)

func main() {
	// 方案一：在同一个位置实现进度打转用英文的"-"
	// 方案二：一般进度打转，一边进度条向右延申，用中文的"—"
	//str := `-\|/`
	// 用于方案二
	str := `—\|/`
	for {
		for _, s := range str {
			fmt.Printf("%c", s)
			time.Sleep(3e8)
			fmt.Printf("\b \b")
		}
	}
}

File: 九、奇淫巧计类/7.实现CMD下载时的进度显示.py
import time


if __name__ == "__main__":
    string = "—\|/"
    n = 45
    while n > 0 :
        for i in range(0, len(string)):
            print(string[i], end="")
            time.sleep(0.3)
            print("\b \b", end="")
